iceoryx_utils/include/iceoryx_utils/internal/concurrent/smart_lock.hpp

```diff
@@ -67,8 +67,16 @@ class smart_lock
     };
 
   public:
+    ///@brief c'tor creating empty smart_lock
     smart_lock() noexcept;
+
+    ///@brief c'tor constructing the underlying object via copy/move
     smart_lock(const T& t) noexcept;
+
+    ///@brief c'tor forwarding all args to the underlying object
+    template <typename... ArgTypes>
+    smart_lock(ArgTypes&&... args) noexcept;
```

Comment:
I guess the `smart_lock` has no tests which could be extended, right?

Reply:
Unfortunately not yet, that's something for a "increase coverage" issue.